Overhaul vignettes and fix most important checks (#56)
52North/sos4R: DEV-README.md

# sos4R developer documentation

This file contains information for developers of the [R](http://r-project.org/) package `sos4R`.

> Documentation for usres can be found in the package's vignette (see `browseVignettes("sos4R")`).

Reviewer: ```suggestion
Documentation for users can be found in the package's vignette (see `browseVignettes("sos4R")`).
```

> <!-- part of this page previously published at https://wiki.52north.org/Geostatistics/Sos4R -->

Reviewer: Better provide a speaking link with some explanation like: 
```suggestion
[Ancient documentation](https://wiki.52north.org/Geostatistics/Sos4R) is available and should be considered if something is missing in this readme or before suggesting changes.
```

Author: Please the the wiki page - it is empty, though of course users could check the wiki history. I think that is unlikely.

## Contribute

Please get in touch with the [community contact](https://52north.org/research/rd-communities/geostatistics/)
of the geostatistics community and read the [Get Involved page](https://52north.org/software/get-involved/) if you want to *become a contributor* -- contributions are welcome!

> You can also dive right in and joint the chat room: [![Join the chat at https://gitter.im/52North/sos4R](https://badges.gitter.im/52North/sos4R.svg)](https://gitter.im/52North/sos4R)

Reviewer: Are you willing and is it possible for you to be always/often available in this gitter chatroom?
```suggestion
You can also dive right in and join the chat room: [![Join the chat at https://gitter.im/52North/sos4R](https://badges.gitter.im/52North/sos4R.svg)](https://gitter.im/52North/sos4R)
```

### How to add an encoder

...

## Releases

A new release shall be uploaded to CRAN after testing and under the following procedure:

- Run the [tests](#tests)
- Check the CI status
  - https://travis-ci.org/52North/sos4R
  - https://ci.appveyor.com/project/nuest/sos4r
- Update version and date in `man/sos4R-package.Rd`
- Update version in `DESCRIPTION`

> - Create a git tag with the version number

Reviewer: ```suggestion
- Create a git tag with the version number in the format `vx.y.z`, e.g `v3.5.56` (following the rules of [semantic versioning](https://semver.org/)
```

Author: Good idea to give an example, semantic version is linked to above already, slightly rephrased it.

- Update NEWS file based on latest commits
- Read and follow http://cran.r-project.org/web/packages/policies.html and http://r-pkgs.had.co.nz/release.html#release-check again
- Run checks

>   - In the parent directory of the project: `R CMD build sos4R; R CMD check --as-cran sos4R_0.<version number>.tar.gz` should have no errors, warnings, or notes

Reviewer: ```suggestion
  - In the parent directory of the project: `R CMD build sos4R; R CMD check --as-cran sos4R_<version number>.tar.gz` should have no errors, warnings, or notes
```
Or is the leading `0.` an CRAN requirement?
